### DISCUSSION TITLE
Armor/tools storage config request.

### DISCUSSION THREAD

**Discussion by Erinbjorn** [BeansGalaxy/Beans-Backpacks-2#124] [Feedback] [answered]
I really like your mod, especially its completely original and realistic behaviour from all the other backpack mods, but without the option to store more than 4  tools or armor its essentially just a big bundle. It would be nice if you could add support that, lets say 1 tool/armor would take up 16 slots of space - making it marginally more viable when using in big modpacks where you loot a lot of armor and tools.

I know I could use another mod along with this, that would allow me to store armor, but those are usually pretty overpowered as they allow you to carry entire sets in a single slot via some special item, which compared to your mod isnt immersive at all. 

Also just wanted to give you a headsup that your mod crashes when using the "you're in grave danger" mod on death, but other grave mods seem to work fine so no big deal.

**Comment by BeansGalaxy** [comment]
I plan on re-writing my config code but I could include an option to let you attach a specific weight to items. I already do this with Enchanted Book and Backpack items where if stored in a backpack they're treated as if they stack to 16. I'll include it with that update since I'll be re-writing everything already.

And most grave mods didn't work with my mod until a few updates ago but "You're In Grave Danger" had special support for my mod. In the update that fixed all the mods it broke YIGD and it's nothing I can really do on my end except feel back abt it. I sent him a message letting him know, it just comes down to when he gets the time for it.

**Reply by Erinbjorn** [reply]
Yea no worries, as long as there are alternatives it'll work. Really glad to see you're responding to ideas, I hope the development goes well for you.

**Reply by BeansGalaxy** [reply]
I started making this mod after the dev of Inventorio said he'll only be updating his mod and not adding any new ideas which I had ***A LOT***. I really only play this mod in a single player creative test world with no other mods and I have some suspicions that most players don't play like that so I'm always looking for more ways to balance my mod.

**Reply by Erinbjorn** [reply]
I see. I mean usuall you wouldnt carry around a lot of armor, weapons and tools as you can craft them, but most players use mods such as Gearifiers and Tiered which add random bonuses and penalties to such items, and usually in dungeons you come across ones with rarer modifiers which have insanely low chance to appear when crafting said item, so you really want to bring them along when you find them - hence my problem with main inv cluttered with armor and tools

**Reply by BeansGalaxy** [reply]
You sound like you have the most blood boiling and intense mod pack. I'll see what I can do with the config and I'll think about a quicker way to access your items and get back to this post.

**Reply by Erinbjorn** [reply]
Lmao not really, just has a lot of dungeons so I find myself in enclosed space, underground and in darkness surrounded by mobs pretty often. Thank you for the consideration

**Reply by Erinbjorn** [reply]
While we're at it just wanna point out that this guy literally described your mod as "being great for storing weapons, tools and non-stackable items" 💀https://youtu.be/BF8TeeWJHyE?si=xmZ_DGfLSSzHv_dJ

**Reply by BeansGalaxy** [reply]
I'm considering just changing over the base functionality to stack everything to 64 in a backpack but it comes with some problems like it's unintuitive and would heavily change the balance of the backpacks. I have been considering raising the size of the backpacks for a while since if you play with biome mods the backpack gets full very quickly and the more mods you have the less balanced the backpacks become.

I keep going back and forth about this but I think this would be a good way to raise the sizes of the backpacks without actually raising them. If not it'll be a easy button in the config to enable.

**Reply by Erinbjorn** [reply]
Nah at that point I'd rather you just leave the mod as it is. Non-stackables beng expensive to carry unless you wear them is a core incentive for the player to create the best possible loadout through enchanting, upgrading and generally maintaining their gear.

Making everything stackable to 64 would imo make it just as unbalanced as other backpack mods, while only creating more problems as you can't even stack the same item with different NBT data like durability, which would result in your backpack being completely littered...

Even if you dont change the "weight" of the items as I first suggested there are still other means of transport - your mod doesnt have to do EVERYTHING. My idea was just that it would make your mod more appealing to people with large modpacks with a lot of tools and other non-stackables, making them more likely to choose your mod over the others while still being much more balanced.

For example I already found out I can use the _carry on_ mod to put my armor on an armor stand, which I can then saddle on a horse with me or a towed cart. So yeah, you definitely shouldn't pressure yourself into significantly changing your mod - there are always other solutions and that's alright

**Reply by BeansGalaxy** [reply]
I gave Decorated Pots and Cauldrons functionality for problems like this though. Pots are good for moving enchanted books and cauldrons are good for moving buckets. If only there was a tool rack item or something like that you could put on that specifically carries armor and tools. The only items I can think of would be a armor stand or anvil but they don't make since for bulk armor. If you have any ideas shoot but I'm trying to avoid adding new items just expand the capabilities of existing ones.

**Comment by BeansGalaxy** [comment, marked as answer]
you can now edit the weight of items through config! Load up your world once to generate the new config and you'll see `item_weight_override`. By default I change Enchanted Books as if each book was 4 items and by typing "all" you change every item in the game. Later I want to add support for tags but you can definitely brute force anything for now.

> If you're adding modded items use their namespace so an entry would look like `"create:brass": 16`

**Reply by Erinbjorn** [reply]
Thank you so much. You're a beast man

**Reply by BeansGalaxy** [reply]
Photo of me -> 🦍
